In [4]:
import pandas as pd
import numpy as np

data = pd.DataFrame({'사회과학':[16,30,12],
                    '자연과학':[12,20,3],
                    '공학':[18,13,14]
                    },index=['4.5~3.5','3.5~2.5','2.5~1.5'])
data


사회과학  자연과학  공학
4.5~3.5    16    12  18
3.5~2.5    30    20  13
2.5~1.5    12     3  14

1. 귀무가설, 연구가설을 작성하시오.

H0 : 학과에 따른 성적의 분포는 동일하다. 

H1 : 동일하지 않다.

2. 학과와 성적이 독립일 때 기댓값을 구하시오.

3. 검정 통계량을 구하고, 연구가설 채택 여부를 작성하시오.

In [6]:
from scipy.stats import chi2_contingency

chi, p,df, expect = chi2_contingency(data)
print('Statistic : ',chi)
print('p-value : ',p)
print('df : ',df)
print('expect : \n',expect)

Statistic :  10.199441509990177
p-value :  0.03719883770303157
df :  4
expect : 
 [[19.33333333 11.66666667 15.        ]
 [26.47826087 15.97826087 20.54347826]
 [12.1884058   7.35507246  9.45652174]]


p-value값이 0.037로 유의수준 0.05보다 작아 귀무가설을 기각하고 대립가설을 채택한다. 즉, 성적의 분포는 유의수준 0.05하에서 학과에 관계없이 동일하지 않다고 판단